---
title: "01-data_import"
output: html_notebook
toc: true
---

```{r}
#| warning: false


library(dplyr)
library(magrittr)






# 1.1. Import metadata


# Set directories
data_dir <- "C:/Users/User/Documents/UCSF/BREATH2.Gene/data"
secure_data_dir <- "C:/Users/User/Documents/UCSF/secure_data"
results_dir <- "C:/Users/User/Documents/UCSF/BREATH2.Gene/results"

# Import metadata
metadata_df_ALL <- read.csv(
  file.path(data_dir, "sample_metadata.csv"),
  stringsAsFactors=F)

JCI_expanded_metadata <- read.csv(
  file.path(secure_data_dir, "fixed_metadata_10242025.csv"),
  stringsAsFactors=F)

# Import ONLY patient IDs from proteomics file
protein_patient_ids <- read.csv(
  file.path(secure_data_dir, "vap_soma_normalized.csv"),
  stringsAsFactors=F) %>%
  dplyr::select(patient_id) %>%
  dplyr::pull(patient_id)






# 1.2. Ready metadata table for LASSO


# Include only patients with both transcriptomic and proteomic data
patients_vector <- intersect(
  protein_patient_ids,
  JCI_expanded_metadata$SubjectID)

# Subset metadata to matched patients
Meta.Data <- JCI_expanded_metadata %>%
  filter(SubjectID %in% patients_vector)

# Ensure ordering matches patient vector
Meta.Data <- Meta.Data[match(patients_vector, Meta.Data$SubjectID), ]

# Build patient info table
patient_info_ALL <- data.frame(
  sample_name = Meta.Data$sanitized_name,
  patient_id = Meta.Data$SubjectID,
  sex = Meta.Data$sex,
  age_in_years = Meta.Data$age_in_years,
  LRTI_adjudication = Meta.Data$LRTI_adjudication,
  stringsAsFactors=F)

# Restrict metadata_df to same patients used in patient_info
metadata_df <- metadata_df_ALL %>%
  filter(sample_name %in% patient_info_ALL$sample_name) %>%
  filter(LRTI_adjudication %in% c("Definite", "No Evidence")) %>%
  left_join(
    patient_info_ALL %>%
      dplyr::select(sample_name, patient_id),
    by = "sample_name"
  ) %>%
  relocate(patient_id, .after = sample_name)

# Write clean metadata file
write.csv(
  metadata_df,
  file.path(data_dir, "metadata_df.csv"),
  row.names=F)






# 1.3. Define workflow parameters


# Define parameters
Definite <- table(patient_info_ALL$LRTI_adjudication)["Definite"]
No_Evidence <- table(patient_info_ALL$LRTI_adjudication)["No Evidence"]
npatients <- Definite + No_Evidence
nfolds <- 5
ntrees <- 10000

# Store parameters
Parameters <- data.frame(c(npatients, Definite, No_Evidence, nfolds, ntrees))
row.names(Parameters) <- c("Patients", "Definite", "No Evidence", "Folds", "RF Trees")
colnames(Parameters) <- "Count"

write.csv(
  Parameters,
  file.path(data_dir, "parameters.csv"),
  row.names=T)






# 1.4. Safety Check


print(Parameters)  

nrow(metadata_df)  # should be 105

sum(duplicated(metadata_df$sample_name))  # Should be 0

sum(is.na(Meta.Data$SubjectID))  # Should be 0






```
